skada/datasets/_office_home.py

```diff
@@ -0,0 +1,338 @@
+    dataset_dir = os.path.join(data_home, loader_spec.dataset_dir)
+    if not os.path.exists(dataset_dir):
+        if not download_if_missing:
+            raise OSError("Data not found and `download_if_missing` is False")
+        os.makedirs(dataset_dir)
+        # juggling with `extract_root` is only required because SURF features
+        # were archived with root folder and DECAF without it
+        _download_office_home(
+            loader_spec.remote,
+            data_home,
+            data_home if loader_spec.extract_root else dataset_dir,
+        )
+
+    (X, y) = _load_office_home(
+        dataset_dir,
+        domain,
+        shuffle=shuffle,
+        random_state=random_state,
+    )
+
+    dataset = DomainAwareDataset()
+    dataset.add_domain(X, y, domain.value)
+
+    return (X, y) if return_X_y else dataset
+
+
+def _download_office_home(remote_spec: RemoteFileMetadata, download_dir, extract_dir):
+    _logger.info(f"Downloading OfficeHome from {remote_spec.url} to {download_dir}")
+    started_at = time.time()
+    dataset_path = _fetch_remote(remote_spec, dirname=download_dir)
+    finished_at = time.time() - started_at
+    _logger.info(f"Finished downloading in {finished_at:0.2f} seconds")
+
+    with zipfile.ZipFile(dataset_path, "r") as zip_ref:
+        zip_ref.extractall(extract_dir)
+
+    os.remove(dataset_path)
+
+
+def _load_office_home(
+    dataset_dir: Union[os.PathLike, str],
+    domain: OfficeHomeDomain,
+    shuffle: bool = False,
+    random_state: Union[None, int, np.random.RandomState] = None,
+) -> Bunch:
+    fullpath = Path(dataset_dir)
+    csv_path = str(fullpath) + "/" + "OfficeHomeResnet50.csv"
+
+    (X, y, domains) = parse_office_home_csv(csv_path)
+
+    # To select only one domain
+    domain_mask = np.char.lower(domains) == domain.value
+    X = X[domain_mask]
+    y = y[domain_mask]
+
+    if shuffle:
+        X, y = shuffle_arrays(X, y, random_state=random_state)
+
+    return X, y
+
+
+def _load_office_home_all(
+    dataset_dir: Union[os.PathLike, str],
+    shuffle: bool = False,
+    random_state: Union[None, int, np.random.RandomState] = None,
+) -> Bunch:
+    fullpath = Path(dataset_dir)
+    csv_path = str(fullpath) + "/" + "OfficeHomeResnet50.csv"
+
+    (X, y, domains) = parse_office_home_csv(csv_path)
+
+    if shuffle:
+        X, y, domains = shuffle_arrays(X, y, domains, random_state=random_state)
+
+    return X, y, domains
+
+
+def parse_office_home_csv(csv_path):
+    # Initialize lists to store features, labels, and domains
+    features = []
+    labels = []
+    domains = []
+
+    with open(csv_path, newline="") as csvfile:
+        # Create a CSV reader object
+        reader = csv.reader(csvfile, quotechar='"')
+
+        # Skip the header row
+        next(reader)
+
+        # Iterate over each row in the CSV file
+        for row in reader:
+            # Append the feature (first column) to the features list
+            features.append(eval(row[0]))
```

Comment:
Oh, performing `eval` on the downloaded file is a huge security risk. Why do we need it here?

Reply:
First of all thanks for all the comments 🙏, its always very helpful. But yes basically when I generated the csv file (by doing pd.to_csv()), my the features in my csv were always `str`. So I figured out I could just do an eval, but ok let me take another look at how I generate the csv.

Reply:
If you use `to_csv()` on the Pandas data frame, maybe `read_csv()` would work to reconstruct it back? 🤔

Reply:
I agree but pandas is not in the requirements.txt of skada and as far as I know we don't want it. Basically what I'm planning to do is to use a csv with 2048 + 2 columns for features + label/domain, that way I think I won't have to use eval()

Reply:
Ah, your are right. We removed pandas from dependencies. Completely forgot about this.